server/src/com/thoughtworks/go/config/update/AgentsEntityConfigUpdateCommand.java

```diff
@@ -141,6 +154,41 @@ public void update(CruiseConfig preprocessedConfig) throws Exception {
         }
     }
 
+    public boolean isPendingAgentUpdated(List<String> uuids, ArrayList<String> allPendingAgentUUIDs) {
+        for (String uuid : uuids) {
+            if (allPendingAgentUUIDs.contains(uuid)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public boolean isInvalidOperationRequested() {
+        return !(resourcesToAdd.isEmpty() && resourcesToRemove.isEmpty() && environmentsToAdd.isEmpty() && environmentsToRemove.isEmpty() && state.isTrue());
+    }
+
+    private void handlePendingAgentsUpdate(ArrayList<String> allPendingAgentUuids, LocalizedOperationResult result, AgentInstances agentInstances, CruiseConfig preprocessedConfig) throws InvalidPendingAgentOperationException {
+        if (isInvalidOperationRequested()) {
+            result.badRequest(LocalizedMessage.string("PENDING_AGENT_INVALID_OPERATION", allPendingAgentUuids));
+            throw new InvalidPendingAgentOperationException(allPendingAgentUuids);
+        }
+        for (String agentUuid : allPendingAgentUuids) {
+            AgentInstance agent = agentInstances.findAgent(agentUuid);
+            agent.enable();
```

Comment:
there is no way to disable the agent, it is directly enabling irrespective of the state being passed.